src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -88,11 +85,31 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
     }
 }
 
-function createDirectInvokeSamDebugConfigurationFromTemplate(
+export function parseCloudFormationResourcesFromTemplate(
```

Comment:
I would change this from a callback to an explicit result. It would allow you to do create a list using `.map(...)` here and in other places rather than pushing to the array.

Callbacks are typically reserved for the observer pattern, where you are registering some logic to be invoked in response to an event. They were also used before Promises (registering some code to run after the completion of an asynchronous bit of code) but Promises and async/await removed the need for them.